### ISSUE TITLE
Maybe field type doesn't respect null check

### ISSUE THREAD

**Issue by amasad** [facebook/flow#207]
```js
/* @flow */
 
// This works:
class Foo {
  test(n: ?number): boolean {
    if (n == null) {
      return false;
    } else {
      return n > 10;
    }
  }
}
 
// This doesn't work:
class Foo {
  n: ?number;
 
  test(): boolean {
    if (this.n == null) {
      return false;
    } else {
      return this.n > 10;
    }
  }
}
```

A workaround:
```js
/* @flow */
 
class Foo {
  n: ?number;
 
  test(): boolean {
    var n = this.n;
    if (n == null) {
      return false;
    } else {
      return n > 10;
    }
  }
}
```

**Comment by gabelevi** [comment]
Which version of flow are you using? I believe this should be fixed by @bhosmer in 0.1.3. Non local variable refinement works slightly differently, but I believe he's handled this case.

I do need to publish an updated binary for 0.1.3...that is being tracked by https://github.com/facebook/flow/issues/206

**Comment by thomasboyt** [comment]
I can confirm that this appears to be working in 0.1.4 :+1:
